### PR TITLE
Storefront footer

### PR DESCRIPTION
**Pull request by ZhangCreations**
## Description

Created footer component for storefront. Has no functionality or links yet.

Link to ticket: https://www.notion.so/uwblueprintexecs/Storefront-footer-component-e4b40a720c904aaf949890e6c49537c8

Screen Shot 2021-03-13 at 16 37 10

### CODE REVIEW

File: src/components/Storefront/Footer.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
+import React, { useState } from 'react';
+import { Image, Center, Text, Link, Button, HStack, VStack, Input } from '@chakra-ui/react';
+import { ChevronRightIcon } from '@chakra-ui/icons';
+import { FaFacebookSquare, FaInstagramSquare, FaYoutube, FaLinkedin, FaTwitter } from 'react-icons/fa';
+import smallLogo from '../../assets/building_up_footer_small_icon.png';
+
+const Footer = props => {
+  const { subscribtionEmail, changeSubscriptionEmail } = useState();
```
Comment (jzxhuang) [inline src/components/Storefront/Footer.js]: typo `subscribtionEmail`

standard react naming convention is `{ xxx, setXxx } = useState()`

File: src/components/Storefront/Footer.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+  const subscribe = e => {
+    e.preventDefault();
+    changeSubscriptionEmail(e);
+  };
+
+  const updateSubscriptionEmail = e => {
+    e.preventDefault();
+    alert(subscribtionEmail);
+  };
```
Comment (jzxhuang) [inline src/components/Storefront/Footer.js]: pretty sure these event handlers are being mapped to the wrong event.

clearer naming may also help: `handleSubscriptionEmailChange` and `handleSubscriptionEmailSubmit` or equivalent

File: src/components/Storefront/Footer.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+  return (
+    <VStack spacing="0px">
```
Comment (jzxhuang) [inline src/components/Storefront/Footer.js]: a `VStack` without spacing is pointless — just use a `Box`

File: src/components/Storefront/Footer.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+      <HStack backgroundColor="black" w="100%" display="flex" paddingTop="54px" paddingBottom="84px">
```
Comment (jzxhuang) [inline src/components/Storefront/Footer.js]: `HStack` by default are flexboxes

File: src/components/Storefront/Footer.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        <Image src={smallLogo} alt="Small Icon" marginLeft="41px" marginRight="47px" alignSelf="flex-start" />
+        <HStack backgroundColor="black" w="100%" display="flex" justifyContent="space-between">
+          <VStack w="13%" alignItems="flex-start" spacing="26px">
```
Comment (jzxhuang) [inline src/components/Storefront/Footer.js]: would prefer not to manually set the width here, just leave the width to be set automatically by the width of the children

File: src/components/Storefront/Footer.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+            <HStack spacing="5">
+              <FaFacebookSquare fontSize="26px" color="white" />
+              <FaInstagramSquare fontSize="26px" color="white" />
+              <Center h="22px" w="22px" backgroundColor="white" borderRadius="2px">
+                <FaYoutube />
+              </Center>
+              <FaLinkedin fontSize="26px" color="white" />
+              <Center h="22px" w="22px" backgroundColor="white" borderRadius="2px">
+                <FaTwitter />
+              </Center>
```
Comment (jzxhuang) [inline src/components/Storefront/Footer.js]: do you know why these two have to be styled differently from the ones above?

File: src/components/Storefront/Footer.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+      <HStack
+        justifyContent="space-between"
+        fontFamily="Karla"
+        color="white"
+        backgroundColor="#BE2A1B"
+        w="100%"
+        paddingTop="31px"
+        paddingBottom="26px"
+      >
+        <Text marginLeft="83px">2020 RAISING THE ROOF. ALL RIGHTS RESERVED. CHARTIABLE #139744569RR0001. </Text>
```
Comment (jzxhuang) [inline src/components/Storefront/Footer.js]: i recommend asking the designers why the `left` and `right` margins aren't equal

**Review by jzxhuang — COMMENTED** [review]
looking good so far!

Would recommend sticking this somewhere that inherits the `ChakraProvider` for the storefront for now, see #38.

Try to avoid inline styling and repeated styilng as much as possible. For example, if you have multiple `marginRight=105px`, put it on the outermost `Box` as `paddingRight=105px`

There's a lot of `letterSpacing` and `fontFamily` set inline, these should typically be set in the global Chakra theme file (i.e. for all `Text` components) rather than on an individual component basis

You can put in the links as much as possible by referencing https://raisingtheroof.org. Let's remove `Sitemap` entirely though. You don't need to implement the email subscription functionality yet. `Site Credits` could link to Blueprint homepage for now

File: src/components/Storefront/Footer.js (hunk 6 above)
Comment (ZhangCreations) [reply src/components/Storefront/Footer.js]: So for those 2, the designer chose to not use the default square icon, but instead the no-square icon and put a box around withy with rounded edges.

Comment (jzxhuang) [reply src/components/Storefront/Footer.js]: let's just use the square-icon for all for consistency

File: src/components/Storefront/Footer.js (hunk 1 above)
Comment (jzxhuang) [reply src/components/Storefront/Footer.js]: u didn't correct the typo 😂 

subscription

File: src/components/Storefront/Footer.js [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+  };
+
+  const handleSubscriptionEmailSubmit = e => {
+    e.preventDefault();
+    alert(subscribtionEmail);
+  };
+
+  return (
+    <Box color="white">
+      <HStack backgroundColor="black" paddingLeft="41px" paddingTop="54px" paddingBottom="84px" paddingRight="105px">
```
Comment (jzxhuang) [inline src/components/Storefront/Footer.js]:
```suggestion
      <HStack backgroundColor="black" py="80px" px="105px">
```

I picked `80px` for vertical padding cuz i think it looks ok 🤷

File: src/components/Storefront/Footer.js [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+          <VStack alignItems="flex-start" spacing="26px" h="100%" marginTop="0px">
+            <Text fontWeight="extrabold" letterSpacing="wider" color="white" fontFamily="Jost">
+              GET IN TOUCH
+            </Text>
+            <Text>
+              116 Industry Street <br /> Suite 313 <br /> Toronto, ON M6M 4L8
+            </Text>
+            <Link href="mailto:info@raisingtheroof.org">info@raisingtheroof.org</Link>
+            <HStack spacing="20px">
+              <FaFacebookSquare fontSize="26px" />
```
Comment (jzxhuang) [inline src/components/Storefront/Footer.js]: wrap icons in Links

make sure the links open in a new tab

File: src/components/Storefront/Footer.js [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+        justifyContent="space-between"
+        backgroundColor="#BE2A1B"
+        paddingLeft="83px"
+        paddingTop="31px"
+        paddingBottom="26px"
+        paddingRight="105px"
+      >
+        <Text>2020 RAISING THE ROOF. ALL RIGHTS RESERVED. CHARTIABLE #139744569RR0001. </Text>
+        <HStack spacing="48px">
+          <Link>PRIVACY POLICY </Link>
```
Comment (jzxhuang) [inline src/components/Storefront/Footer.js]: pls link to thte actual RTR pages

File: src/components/Storefront/Footer.js [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+          </VStack>
+        </HStack>
+      </HStack>
+      <HStack
+        justifyContent="space-between"
+        backgroundColor="#BE2A1B"
+        paddingLeft="83px"
+        paddingTop="31px"
+        paddingBottom="26px"
+        paddingRight="105px"
```
Comment (jzxhuang) [inline src/components/Storefront/Footer.js]: same padding comment as way above

File: src/components/Storefront/Footer.js [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+} from 'react-icons/fa';
+import smallLogo from '../../assets/building_up_footer_small_icon.png';
+
+const Footer = () => {
+  const [subscribtionEmail, setSubscribtionEmail] = useState('');
+
+  const handleSubscriptionEmailChange = e => {
+    e.preventDefault();
+    setSubscribtionEmail(e.target.value);
+    console.log(subscribtionEmail);
```
Comment (jzxhuang) [inline src/components/Storefront/Footer.js]: let's remove this console log

**Comment by jzxhuang** [comment]
@ZhangCreations can you also add 

```
            <Route exact path="/footer">
              <ChakraProvider theme={storeTheme}>
                <Footer />
              </ChakraProvider>
            </Route>
```

in `App.js` under the route `chakraExpoStore`

this will be temporary but let's add it for now, thanks
image

**Review by jzxhuang — APPROVED** [review]

**Pull request merged**